### ISSUE TITLE
[Feature Request] Add `indextree!` macro to quickly construct an `Arena`

### ISSUE THREAD

**Issue by alexmozaidze** [saschagrunert/indextree#109]
I often find myself constructing several AST's and then testing different operations on them. Having a quick one-liner to construct a tree (`Arena`) would be most useful.
```rust
#[cfg(test)]
mod tests {
    lazy_static! {
        static ref TEST_SUBJECT1: Arena<&str> = indextree! {
            "1" => {
                "1_1",
                "1_2" => { "2_1" },
                "1_3",
            },
            // etc.
        };
    }
}
```

I could try to implement it myself, but that'll take a while, since I've only ever created declarative macros as opposed to procedural macros.

**Comment by alexmozaidze** [comment]
Fixed with #110

**Comment by alexmozaidze** [comment]
Would be nice, yeah. Other than refactoring tests, I don't think there's much to do in order to improve indextree for my particular use-case.
